metadata_service/api/table.py

```diff
@@ -188,16 +204,31 @@ def put(self, table_uri: str, tag: str) -> Iterable[Union[Mapping, int, None]]:
         :param tag:
         :return:
         """
+        args = self.parser.parse_args()
+        # use tag_type to distinguish between tag and badge
+        tag_type = args.get('tag_type', 'default')
         try:
-            self.client.add_tag(table_uri=table_uri, tag=tag)
-            return {'message': 'The tag {} for table_uri {} '
+            self.client.add_tag(table_uri=table_uri,
+                                tag=tag,
+                                tag_type=tag_type)
+            return {'message': 'The tag {} for table_uri {} with type {} '
                                'is added successfully'.format(tag,
-                                                              table_uri)}, HTTPStatus.OK
+                                                              table_uri,
+                                                              tag_type)}, HTTPStatus.OK
         except NotFoundException:
             return \
-                {'message': 'The tag {} for table_uri {} '
+                {'message': 'The tag {} for table_uri {} with type {} '
                             'is not added successfully'.format(tag,
-                                                               table_uri)}, \
+                                                               table_uri,
+                                                               tag_type)}, \
+                HTTPStatus.NOT_FOUND
+        except BadgeNotInWhitelistException:
```

Comment:
How about validating whitelist in the API level? So that it won't need to be implemented on multiple proxies?

Reply:
sg